ArduCopter/mode_poshold.cpp

```diff
@@ -116,15 +116,20 @@ void ModePosHold::run()
         break;
 
     case AltHoldModeState::Landed_Ground_Idle:
-        loiter_nav->clear_pilot_desired_acceleration();
-        loiter_nav->init_target();
-        attitude_control->reset_yaw_target_and_rate();
+        attitude_control->reset_yaw_target_and_rate(false);
+        attitude_control->reset_rate_controller_I_terms_smoothly();
         init_wind_comp_estimate();
-        FALLTHROUGH;
+        break;
 
     case AltHoldModeState::Landed_Pre_Takeoff:
+        if (!motors->using_hdg_error_correction()) {
+            attitude_control->reset_target_and_rate(false);
+        }
         attitude_control->reset_rate_controller_I_terms_smoothly();
+        init_wind_comp_estimate();
         pos_control->relax_z_controller(0.0f);   // forces throttle output to decay to zero
+        loiter_nav->clear_pilot_desired_acceleration();
+        loiter_nav->init_target();
```

Comment:
@MattKear @lthall This does not follow the same flow as Loiter.  I changed it because loiter is done this way.  Not sure why they would be different.  Again I can change it back if you think it should not change.  Or I can just do a separate PR.

Reply:
fixed.  Thanks,